### ISSUE TITLE
Ambiguous Handle name

### ISSUE THREAD

**Issue by miyconst** [Starcounter/Starcounter#2733]
There is a static class `Starcounter.Handle` which has set of methods and properties. The client side events in `Starcounter.Json` class are handled via `void Handle(Input.Xxxx Action)` methods.

That is why calls like `Handle.CallLevel` are invalid inside `Starcounter.Json` viewmodels which has at least one `void Handle` definition. The full name should be used instead: `Starcounter.Handle.CallLevel`.

I suggest to rename one of the items, this also will remove ambiguous when discussing things.

@warpech, @alemoi, @chrhol what do you think?

**Comment by alemoi** [comment]
My input on this: I think that Handlers and Handles are too similar names and its hard to talk about these two in distinguishable way. Maybe Handles should change the name?

**Comment by chrhol** [comment]
I agree with @alemoi, the input-handler should be renamed. The question is to what :) I will ping @Starcounter-Jack as well so we can decide on how we want the public API to look.

**Comment by miyconst** [comment]
I would like to see something more traditional. 

```json
{
    "Html": "/LoginPage.html",
    "Username$": "",
    "Password$": "",
    "SubmitClick$": 0
}
```

```cs
partial class LoginPage : Page {
    public LoginPage() {
        //Anonymous method
        this.Input.Password += action => { };

        //Normal methods
        this.Input.Username += this.OnUsernameChange;
        this.Input.SubmitClick += this.OnSubmitClick;
    }

    protected void OnSubmitClick(InputAction Action) {
    }

    protected void OnUsernameChange(InputAction Action) {
    }
}
```

This approach would provide intellisense for every part, and it also wold be possible to discover how to  subscribe for model changes without actually reading the docs.

**Comment by per-samuelsson** [comment]
Whats the current approach, @miyconst? Is it [this](http://starcounter.io/guides/web/server-side-view-models/) (just using new/alternative naming, i.e. Page != View)?

**Comment by miyconst** [comment]
Page is almost a `Json`, just with some nice features for htmlmerger.
https://github.com/Starcounter/Starcounter/blob/21e3eccd53af2c80a9945d5e09a66da93251bb7e/src/Starcounter/Page.cs

**Comment by miyconst** [comment]
The new part in my approach is `Input.X` events, not the `Page` class and view-model itself.

**Comment by per-samuelsson** [comment]
> The new part in my approach is Input.X events

Yeah, that much even I understood. I compared it to `Handle(Input.[Class] input)` in the referenced guide, I just wanted to know that comparison was valid, and now I see it is, when you have provided the whole picture. Thanks.

I'm not sure exactly where I stand on this. I see pros and cons of both initially. Let me get back on it.

**Comment by miyconst** [comment]
Sure. Even if we implement the new approach, we still should not abandon the old one, we need the backwards compatibility.

**Comment by chrhol** [comment]
The main reason for not having these patterns is that the actual instance of the json should be treated as a proxyobject (same as for databaseobjects) and have as few values stored in the instance as possible so they are fast to create and clean up. The instance might also be recreated during different requests (not implemented at this time though) to avoid keeping alot of clrobjects in memory since that will probably not work very well for handling many users.

Today all events and triggers is created once (stored on what we call a `Template`), and the template is then reused and attached to each new instance. Actual values for a jsonobject might also be stored in a different place, depending on how it is used.

We should (of course) improve how you work with TypedJson, but we should also be really careful when deciding how we want it to look and avoid as much as possible on storing events/data per instance.

**Comment by miyconst** [comment]
Okay.. does this all mean that I should avoid creating custom events in my TypedJson classes?

Other possible solutions:

```cs
[LoginPage.Input.SubmitClick]
void OnSubmitClick(InputAction Action) {
}

[LoginPage.Input.SubmitClick]
void CustomOnSubmitClick(InputAction Action) {
}
```

```cs
protected override void HandleChange(InputAction Action) {
	switch (Action.Property) {
		case Input.SubmitClick:
			break;
		case Input.Username:
			break;
		default:
			break;
	}
}
```

Any other ideas are welcome. The points I try to achieve:

1. Intellisense works for TypedJson handles.
2. Project does not build with a wrong handle.
